dukan/src/main/kotlin/net/thechance/dukan/service/DukanShelfService.kt

```diff
@@ -27,4 +31,17 @@ class DukanShelfService(
             )
         )
     }
+    fun deleteShelf(shelfId: UUID, ownerId: UUID) {
+        val dukan = dukanRepository.findByOwnerId(ownerId) ?: throw DukanNotFoundException()
```

Comment:
This logic is repeated I suggest creating a function in DukanService that retrieves a dukan by ID and throws this exception